docs/source/dev/py_postprocessing.rst

```diff
@@ -0,0 +1,178 @@
+.. _development-pytools:
+
+Python postprocessing tool structure
+====================================
+
+For each plugin there are (or have to be implemented) three different classes.
+
+1. A data reader class responsible for loading the data from the simulation directory
+2. A visualizer class that outputs a matplotlib plot
+3. A jupyter-widget class that exposes the parameters of the matplotlib visualizer to the user
+   via other widgets
+
+Data reader
+~~~~~~~~~~~
+The data readers should reside in the ``lib/python/picongpu/plugins/`` directory.
+
+Each reader class needs to implement the following interface functions:
+
+.. code:: python
```

Comment:
Did we forget to write a base class for this? If so, we should build one in a follow-up PR.

Reply:
Another follow-up to this PR:
On Friday, we realized we forgot to add a meta-data method to return which units (and unit dimensions) the read data have. We should add a `get_meta` or similar method. We should make this similar or identical to `unitSI` and `unitDimension` in [openPMD](https://github.com/openPMD/openPMD-standard/blob/1.1.0/STANDARD.md#required-for-each-record-component) and/or can rely on [pint-strings](https://pint.readthedocs.io).

Reply:
Yes, so far we don't have a base class for the readers. Would you want it to contain all the interface functions (and raising `NotImplementedError` or should this class already implement some logic?

Reply:
Implemented in #2730

Reply:
added now with last commit to avoid updating the interface in the manual *and* the source (now source only :) )